### ISSUE TITLE
Duplicating of events doesn't work

### ISSUE THREAD

**Issue by FreneticScribbler** [nottinghamtec/PyRIGS#451]
FFS. 
```
2021-10-26T20:38:51.791606+00:00 app[web.1]:   File "/app/RIGS/models.py", line 347, in display_id
2021-10-26T20:38:51.791606+00:00 app[web.1]:     return str("N%05d" % self.pk)
2021-10-26T20:38:51.791606+00:00 app[web.1]: TypeError: %d format: a number is required, not NoneType
```

**Comment by FreneticScribbler** [comment]
Ahh, ok, pk is null if you hit save and there's an error. Related to #189 I imagine

**Comment by FreneticScribbler** [comment]
Fixed in 014b00bc30feb88b91fd966d3174a73d8d72de8a
